### PR TITLE
cleanup #2034: remove mlog and causalft

### PR DESCRIPTION
**Pull request by ericjbohm**
Message logging and causal ft have been broken for the
past few releases.  Based on that defacto deprecation,
we're removing all the causal and mlog code for 6.10.

Primary implemented via unifdef
unifdef -K -m \-UCK_MULTICAST_MSG_MLOG \-U_FAULT_MLOG_
 \-U_FAULT_CAUSAL_ \-UCK_MULTICAST_MSG_MLOG \-U_MLOG_BCAST_TREE_

Followed by deletion and build system excisions.

### CODE REVIEW

**Comment by ericjbohm** [comment]
CMK_MESSAGE_LOGGING wasn’t on the unifdef line, I’ll submit a revision for that.

My understanding is that FAULT_EVAC is a different feature to trigger migration away. Used by Causal, but could potentially be used by other protocols.

File: src/ck-core/ck.C [hunk 1]
```diff
@@ -1472,7 +1424,7 @@ void _skipCldEnqueue(int pe,envelope *env, int infoFn)
     CmiSetInfo(env,infoFn);
     if (pe==CLD_BROADCAST) {
 #if CMK_MESSAGE_LOGGING
-	if(env->flags & CK_FREE_MSG_MLOG)
+	if(env->flags)
```
[the remark this answers is not in the document]
Comment (stwhite91) [reply src/ck-core/ck.C]: Still needs to be addressed

[the remark this answers is not in the document]
Comment (matthiasdiener) [reply src/ck-core/ck.C]: Seems to be fixed with 404cfb5eab4566d301add2b855b47a155f15125a?

[the remark this answers is not in the document]
Comment (ericjbohm) [reply src/ck-core/ck.C]: Yes, that unifdefs CMK_MESSAGE_LOGGING, to get that block of code along with all the others under that macro.

**Pull request merged**